### PR TITLE
Add VCE(Voice-Control-Engine) guide

### PR DESCRIPTION
**Pull request by sungraejo**
ACR-1175

Signed-off-by: sungrae jo <seongrae.jo@samsung.com>

### CODE REVIEW

File: docs/application/native/guides/text-input/voice-control-engine.md [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+        int ret = VCE_ERROR_NONE;
+
+        ret = vce_start_recording();
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+
+- Stop recording voice using the `vce_stop_recording()` function. If the function call is successful, `vce_request_callback_s` call to the `stop` function in a struct.
```
[the remark this answers is not in the document]
Comment (sungraejo) [reply docs/application/native/guides/text-input/voice-control-engine.md]: I will fixed 466 Line. It's wrong.

Stop recording voice using the `vce_stop_recording()` function. If the function call is successful, `vce_request_callback_s` call to the `stop` function in a struct.
to
Stop recording voice using the `vce_stop_recording()` function. If the function call is successful, You can not receive recording data from `set_recording` function.

File: docs/application/native/guides/text-input/voice-control-engine.md [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+
+Voice-Control-Engine (VCE) is an engine to recognize sound data. The user records this data and sends the result as a predefined command.
+
+VCE developers can provide VCE service users, to apply VCE with functions that is necessary to operate the engine.
+
+The main features of the VCE API include:
+
+- Preparing the VCE service for use
+
+  You can connect the VCE service application. [Basic VCE Processes](#basic) and [Prerequisites](#prerequisites), are defined.
```
Comment (sabeena21) [inline docs/application/native/guides/text-input/voice-control-engine.md]: Please change:
You can connect the VCE service application. [Basic VCE Processes](#basic) and [Prerequisites](#prerequisites), are defined.
to 
You can connect the VCE service application. [Basic VCE Processes](#basic) and [Prerequisites](#prerequisites) are defined.
"Please can you remove comma before and"
Question: Is there any specific reason that the comma is retained in this sentence?

File: docs/application/native/guides/text-input/voice-control-engine.md [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+        int ret = VCE_ERROR_NONE;
+        int g_vce_command;
+
+        ret = vce_get_foreach_command(g_vce_command, command_cb, NULL);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Get command length using the `vce_get_command_count()` function. Get `g_vce_command` data from set_commands()
```
Comment (sabeena21) [inline docs/application/native/guides/text-input/voice-control-engine.md]: Get command length using the `vce_get_command_count()` function. Get `g_vce_command` data from set_commands()
"Please can you add period at the end of the sentence"

File: docs/application/native/guides/text-input/voice-control-engine.md [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+        int ret = VCE_ERROR_NONE;
+        int is_consumed;
+
+        ret = vce_send_result(VCE_RESULT_EVENT_SUCCESS, NULL, 0, "text", NULL, NULL, VC_RESULT_MESSAGE_NONE, &is_consumed, NULL);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Send the Automatic Speech Recognition (ASR) result to the engine service user using the vce_send_asr_result() function.
```
Comment (sabeena21) [inline docs/application/native/guides/text-input/voice-control-engine.md]: Send the Automatic Speech Recognition (ASR) result to the engine service user using the vce_send_asr_result() function.

Please change the sentence:
Send the Automatic Speech Recognition (ASR) result to the engine service user using the `vce_send_asr_result()` function.

File: docs/application/native/guides/text-input/voice-control-engine.md [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+    send_asr_result(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        ret = vce_send_asr_result(VCE_ASR_RESULT_EVENT_FINAL_RESULT, "asr_result", NULL);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Send the Natural Language Generation (NLG) result to the engine service user using the vce_send_nlg_result() function.
```
Comment (sabeena21) [inline docs/application/native/guides/text-input/voice-control-engine.md]: Send the Natural Language Generation (NLG) result to the engine service user using the vce_send_nlg_result() function.
Please change the sentence as:
Send the Natural Language Generation (NLG) result to the engine service user using the `vce_send_nlg_result()` function.

File: docs/application/native/guides/text-input/voice-control-engine.md (hunk 2 above)
Comment (sungraejo) [reply docs/application/native/guides/text-input/voice-control-engine.md]: Done.
Answer : No, it's a misstake. your suggestion looks good. thank you.

File: docs/application/native/guides/text-input/voice-control-engine.md (hunk 3 above)
Comment (sungraejo) [reply docs/application/native/guides/text-input/voice-control-engine.md]: Done.

File: docs/application/native/guides/text-input/voice-control-engine.md (hunk 4 above)
Comment (sungraejo) [reply docs/application/native/guides/text-input/voice-control-engine.md]: Done.

File: docs/application/native/guides/text-input/voice-control-engine.md (hunk 5 above)
Comment (sungraejo) [reply docs/application/native/guides/text-input/voice-control-engine.md]: Done.

File: docs/application/native/guides/text-input/voice-control-engine.md [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+    }
+  ```
+
+
+<a name="control_record"></a>
+## VCE Control Recording
+
+You can control recording of the following information about the VCE:
+
+- Start recording voice using the `vce_start_recording()` function. If the function call is successful, You can receive recording data from `set_recording` function.
```
Comment (sabeena21) [inline docs/application/native/guides/text-input/voice-control-engine.md]: You can receive recording data from `set_recording` function. 

Please change:

you can receive recording data from `set_recording` function.

File: docs/application/native/guides/text-input/voice-control-engine.md [hunk 7]
```diff
@@ -0,0 +1,527 @@
...
+    {
+        int ret = VCE_ERROR_NONE;
+
+        ret = vce_start_recording();
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Stop recording voice using the `vce_stop_recording()` function. If the function call is successful, You can not receive recording data from `set_recording` function.
```
Comment (sabeena21) [inline docs/application/native/guides/text-input/voice-control-engine.md]: You can not receive recording data from `set_recording` function. 
Please change:
 you can not receive recording data from `set_recording` function.

File: docs/application/native/guides/text-input/voice-control-engine.md (hunk 6 above)
Comment (sungraejo) [reply docs/application/native/guides/text-input/voice-control-engine.md]: Done.

File: docs/application/native/guides/text-input/voice-control-engine.md (hunk 7 above)
Comment (sungraejo) [reply docs/application/native/guides/text-input/voice-control-engine.md]: Done.

**Comment by mijin80lee** [comment]
TD Sync Completed.(for Tizen_5.0_prepare)

**Pull request merged**